lib/Dialect/ESI/cosim/cosim_dpi_server/DpiEntryPoints.cpp

```diff
@@ -0,0 +1,207 @@
+//===- DpiEntryPoints.cpp - ESI cosim DPI calls -----------------*- C++ -*-===//
+//
+// Cosim DPI function implementations. Mostly C-C++ gaskets to the C++
+// RpcServer.
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Dialect/ESI/cosim/Server.h"
+#include "circt/Dialect/ESI/cosim/dpi.h"
+
+#include <algorithm>
+#include <cstdlib>
+#include <iostream>
+
+using namespace circt::esi::cosim;
+
+static RpcServer *server = nullptr;
+
+// ---- Helper functions ----
+
+/// Get the TCP port on which to listen. Defaults to 0xECD (ESI Cosim DPI), 3789
+/// in decimal.
+int getPort() {
+  const char *portEnv = getenv("COSIM_PORT");
+  if (portEnv == nullptr) {
+    printf("[COSIM] RPC server port not found. Defaulting to 3789.\n");
+    return 0xECD;
+  }
+  printf("[COSIM] Opening RPC server on port %s\n", portEnv);
+  return std::strtoull(portEnv, nullptr, 10);
+}
+
+/// Check that an array is an array of bytes and has some size.
+// NOLINTNEXTLINE(misc-misplaced-const)
+static int validateSvOpenArray(const svOpenArrayHandle data,
```

Comment:
I don't think this needs to be marked const?

Reply:
These function signatures were generated by an RTL simulator (see dpi.h) so I don't change them to be more rational here. The resulting code gets dynamically linked in to a bunch of different simulations so I'm concerned about maintaining binary compatibility with all the simulators.

I'll add a note to the comment header of this file and dpi.h.

Reply:
ok!